akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/Behaviors.scala

```diff
@@ -170,14 +175,24 @@ object Behaviors {
   /**
    * Construct an immutable actor behavior from a partial message handler which treats undefined messages as unhandled.
    */
-  def immutablePartial[T](onMessage: PartialFunction[(ActorContext[T], T), Behavior[T]]): Immutable[T] =
-    Behaviors.immutable[T] { (ctx, t) ⇒ onMessage.applyOrElse((ctx, t), (_: (ActorContext[T], T)) ⇒ Behaviors.unhandled[T]) }
+  def receivePartial[T](onMessage: PartialFunction[(ActorContext[T], T), Behavior[T]]): Immutable[T] =
+    Behaviors.receive[T] { (ctx, t) ⇒
+      onMessage.applyOrElse((ctx, t), (_: (ActorContext[T], T)) ⇒ Behaviors.unhandled[T])
+    }
+
+  /**
+   * Construct an immutable actor behavior from a partial message handler which treats undefined messages as unhandled.
+   */
+  def receiveMessagePartial[T](onMessage: PartialFunction[T, Behavior[T]]): Immutable[T] =
+    Behaviors.receive[T] { (_, t) ⇒
+      onMessage.applyOrElse(t, (_: T) ⇒ Behaviors.unhandled[T])
+    }
 
   /**
    * Construct an actor behavior that can react to lifecycle signals only.
    */
   def onSignal[T](handler: PartialFunction[(ActorContext[T], Signal), Behavior[T]]): Behavior[T] =
```

Comment:
this is the only `on` in `Behaviors.`
should it be `receiveSignal`?
(hopefully nobody will suggest that we should have one of this without ctx, because then we are in naming trouble)

Reply:
Hehe yeah was thinking about naming during this PR -- I don't think this one needs the 2 versions, just receiveSignal is good

Reply:
There is also `.onSignal` in the `Immutable` that is returned by `receive`. Renaming that to `receiveSignal` might be problematic because there is already `def receiveSignal(ctx: ActorContext[T], msg: Signal)` in ExtensibleBehavior, but perhaps that overload works thanks to different number of parameters.

Reply:
This overload should be fine I think, will try